validator/client/src/main/java/tech/pegasys/teku/validator/client/SignedValidatorRegistrationFactory.java

```diff
@@ -32,57 +32,41 @@
 public class SignedValidatorRegistrationFactory {
   private static final Logger LOG = LogManager.getLogger();
 
-  private final ProposerConfigPropertiesProvider validatorRegistrationPropertiesProvider;
+  private final ProposerConfigPropertiesProvider proposerConfigPropertiesProvider;
   private final TimeProvider timeProvider;
 
   public SignedValidatorRegistrationFactory(
-      final ProposerConfigPropertiesProvider validatorRegistrationPropertiesProvider,
+      final ProposerConfigPropertiesProvider proposerConfigPropertiesProvider,
       final TimeProvider timeProvider) {
-    this.validatorRegistrationPropertiesProvider = validatorRegistrationPropertiesProvider;
+    this.proposerConfigPropertiesProvider = proposerConfigPropertiesProvider;
     this.timeProvider = timeProvider;
   }
 
-  public Optional<SafeFuture<SignedValidatorRegistration>> createSignedValidatorRegistration(
+  public SafeFuture<SignedValidatorRegistration> createSignedValidatorRegistration(
       final Validator validator,
       final Optional<SignedValidatorRegistration> oldValidatorRegistration,
       final Consumer<Throwable> errorHandler) {
-    return createSignedValidatorRegistration(validator, oldValidatorRegistration)
-        .map(registrationFuture -> registrationFuture.whenException(errorHandler));
-  }
-
-  private Optional<SafeFuture<SignedValidatorRegistration>> createSignedValidatorRegistration(
-      final Validator validator,
-      final Optional<SignedValidatorRegistration> oldValidatorRegistration) {
 
     final BLSPublicKey publicKey = validator.getPublicKey();
 
-    final boolean builderEnabled =
-        validatorRegistrationPropertiesProvider.isBuilderEnabled(publicKey);
-
-    if (!builderEnabled) {
-      LOG.trace("Validator registration is disabled for {}", publicKey);
-      return Optional.empty();
-    }
-
-    final Optional<Eth1Address> maybeFeeRecipient =
-        validatorRegistrationPropertiesProvider.getFeeRecipient(publicKey);
-
-    if (maybeFeeRecipient.isEmpty()) {
-      LOG.debug(
-          "Couldn't retrieve fee recipient for {}. Will skip registering this validator.",
-          publicKey);
-      return Optional.empty();
-    }
-
-    final Eth1Address feeRecipient = maybeFeeRecipient.get();
-    final UInt64 gasLimit = validatorRegistrationPropertiesProvider.getGasLimit(publicKey);
+    final Eth1Address feeRecipient =
+        proposerConfigPropertiesProvider
+            .getFeeRecipient(publicKey)
+            .orElseGet(
+                () -> {
+                  LOG.warn(
+                      "Fee recipient not configured for {}. Will use a burn address for the validator registration.",
+                      validator.getPublicKey());
+                  return Eth1Address.ZERO;
```

Comment:
My concern is previously we didn't register such validators, so they were able to produce blocks locally where fee recipient could be configured on beacon node, since this change they will be registered and it could be fee loss

Reply:
I think it's ok. We are on the VC side, and we enforce the presence of a default fee recipient otherwise the client doesn't even start. `validatorRegistrationPropertiesProvider.getFeeRecipient` returns an `Optional` but should never happen in practice.

Reply:
Changed it to throwing an exception in which case only that registration will be skipped with a clear error message. Think it's better.